homeassistant/components/smart_meter_b_route/sensor.py

```diff
@@ -0,0 +1,93 @@
+"""Smart Meter B Route."""
+
+import logging
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfElectricCurrent, UnitOfEnergy, UnitOfPower
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ATTR_API_INSTANTANEOUS_CURRENT_R_PHASE,
+    ATTR_API_INSTANTANEOUS_CURRENT_T_PHASE,
+    ATTR_API_INSTANTANEOUS_POWER,
+    ATTR_API_TOTAL_CONSUMPTION,
+    DOMAIN,
+)
+from .coordinator import BRouteUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_DESCRIPTIONS = (
+    SensorEntityDescription(
+        key=ATTR_API_INSTANTANEOUS_CURRENT_R_PHASE,
+        name="Instantaneous Current R Phase",
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+    ),
+    SensorEntityDescription(
+        key=ATTR_API_INSTANTANEOUS_CURRENT_T_PHASE,
+        name="Instantaneous Current T Phase",
+        device_class=SensorDeviceClass.CURRENT,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+    ),
+    SensorEntityDescription(
+        key=ATTR_API_INSTANTANEOUS_POWER,
+        name="Instantaneous Power",
+        device_class=SensorDeviceClass.POWER,
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPower.WATT,
+    ),
+    SensorEntityDescription(
+        key=ATTR_API_TOTAL_CONSUMPTION,
+        name="Total Consumption",
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Smart Meter B-route entry."""
+    coordinator: BRouteUpdateCoordinator = hass.data[DOMAIN][config.entry_id]
+
+    async_add_entities(
+        SmartMeterBRouteSensor(coordinator, description)
+        for description in SENSOR_DESCRIPTIONS
+    )
+
+
+class SmartMeterBRouteSensor(SensorEntity, CoordinatorEntity[BRouteUpdateCoordinator]):
```

Comment:
All new intergations require to use `_attr_has_entity_name = True`. Should we also add a device to group the entities?